runtime/src/sorted_storages.rs

```diff
@@ -54,9 +54,25 @@ impl<'a> SortedStorages<'a> {
     // 2. slots and source are the same len
     pub fn new_with_slots<'b>(
         source: impl Iterator<Item = (&'a SnapshotStorage, &'b Slot)> + Clone,
+        min_slot: Option<Slot>,
```

Comment:
Nit: Add small comment `A slot used as a lower bound, but potentially smaller than the smallest slot in the given 'source' iterator`